fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt

```diff
@@ -41,12 +67,38 @@ fun DebugOverlay() {
                 .align(Alignment.CenterStart)
                 .padding(start = 16.dp),
             onClick = {
-                scope.launch {
-                    snackbarState.showSnackbar("Show debug drawer")
-                }
+                drawerEnabled = true
             },
         )
 
+        // ModalDrawer utilizes a Surface, which blocks ALL clicks behind it, preventing the app
+        // from being interactable. This cannot be overridden in the Surface API, so we must hide
+        // the entire drawer when it is closed. Whenever the app is Compose from
+        // the root layer, we can remove this.
```

Comment:
It seems fine to keep this even when the app is fully compose, right? I would guess we probably want to skip these layout passes entirely when the drawer is not visible in either case

Reply:
If the app is fully compose, then we I suppose there's two paths for the future of this feature's UI:
1. Make the drawer accessible from the root level and ANY feature can display one (which was kinda what I was thinking by including that comment)
2. Keep the `DebugOverlay` its own distinct layer, even after 100% migration, and have it act independently without any implementation details of what is behind it.

Reply:
I'll update the comment though to not assume any specific direction we take with getting Fenix 100% compose